### ISSUE TITLE
About the precision of example (walkthrough-tensorflow-object-detection)

### ISSUE THREAD

**Issue by bobleegogogo** [developmentseed/label-maker#57]
Hallo there, 
    I have tried witht he example of : A building detector with TensorFlow API at mexico city, I just follow all the default setting of boundary box and training model. Then I got my model trained as well as the test result, the question is that from the blog page" There are 227 buildings in the test dataset, and 191 buildings are predicted correctly by the model (84%)." 
    I was wondering that is this "84%" refer to the precision? If not, what's this number means?  For my own model, I trained it around 20000, and select one of the check point in "training" folders, then got only precision around 20%, so I want to know how we could select the best checkpoint during the training? or we just monitoring the loss and select some checkpoint manually?

Best Regards,

Hao

**Comment by drewbo** [comment]
Hi @bobleegogogo, of the 227 buildings that are in the test dataset, we ended up with bounding boxes roughly matching up with 191 (84%). The default settings produce some false positives which could be filtered out via post-processing but it does lead to a lower calculated precision score during training.

**Comment by bobleegogogo** [comment]
Hi @drewbo, so happy about your fast reply!  Is this 227 building belong to one certain tiles ? or this 227 building measn 227 image tiles? I got little bit confusing here. And you just mentioned about the post-processing, would you guys have some tips like for which kind of post-processing would be possible for precision improvment?

Best
Hao

**Comment by drewbo** [comment]
@bobleegogogo sorry for dropping this thread. Did you get the problem resolved in #67? I don't have a good post-processing method available right now but will be happy to share if we come across better methods in the future

**Comment by bobleegogogo** [comment]
@drewbo I got the problem resolved by modifying the code in tf_records_generation.py myself, actually I think the split of the test and training sample is not a small issue, we have to be 100% sure about the test and training is independent, otherwise the result is not reliable.

**Comment by Geoyi** [comment]
@bobleegogogo, do you create a pull request to submit your code at #67? @drewbo and I can give a check?

**Comment by bobleegogogo** [comment]
yes, I would like to do this.

M.Sc Hao. Li

E-mail: leebobgiser316@gmail.com

Phone:+0049 015908113885


2018-04-18 17:45 GMT+02:00 Zhuangfang Yi <notifications@github.com>:

**Comment by Geo-Trackers** [comment]
I want to detect brick-kilns in satellite imagery. I trained my model using Tensorflow object detection API. Now, I want to run inference over my image tiles. Can i use gdaltools to generate tiles and run inference over each tiles, and stich them back together for visualization over the large area of satellite imagery? Also, if my brick kiln falls in two image tiles, will the script tf_od_predict.py compute non-maximum suppression and get rid of low probability predictions, while i stich the images tiles back together. 
Thank you. I look forward to hear from you.
